app/assets/javascripts/dropzone_image_upload.js

```diff
@@ -22,21 +25,29 @@
 
   var bindHandlers = function() {
     $('.campaign-images').on('ajax:success', "a[data-method=delete]", function(){
-      $(this).parents('.image-thumb').fadeOut();
+      $(this).parents('.dz-preview').fadeOut();
+      var imageId = $(this).parents('[data-image-id]').data('image-id');
+      $.publish('image:destroyed', imageId)
     });
   };
 
-  var checkAnyImages = function() {
-    if( $('.campaign-images img').length == 0) {
-      $('.campaign-images').hide();
-    }
-  };
-
   var initialize = function() {
     configureDropZone();
     bindHandlers();
   };
 
+  var addImageOption = function(e, file, id, html) {
+    var newOption = "<option value='"+id+"'>"+file.name+"</option>";
```

Comment:
Is it possible to also construct these as JQ elements? Would make this a little more readable.